components/raftstore/src/store/read_queue.rs

```diff
@@ -165,9 +165,17 @@ impl ReadIndexQueue {
         T: IntoIterator<Item = (Uuid, u64)>,
     {
         for (uuid, index) in states {
-            assert_eq!(uuid, self.reads[self.ready_cnt].id);
-            self.reads[self.ready_cnt].read_index = Some(index);
-            self.ready_cnt += 1;
+            // Read requests can be out-of-order even on leader.
```

Comment:
How?

Reply:
Suppose a raft group contains 2 peers. Firstly we propose a read index on leader, it will wait a heartbeat response to get ready. Then we remove the another peer by configuration change, then propose a new read index again. Then the second one can be pushed into read states before the first one.
You can get more details in the new added test case.

Reply:
This seems related to tikv/raft-rs#345. I suggest to fix it in raft-rs, otherwise we can allow unexpected bugs succeed successfully.

Reply:
Yes, fix it in raft-rs is better. So let the contributer try it?